tests/unit/amo/components/TestPermissions.js

```diff
@@ -0,0 +1,179 @@
+import { PermissionUtils } from 'amo/components/PermissionsCard/permissions';
+import { createInternalAddon } from 'core/reducers/addons';
+import {
+  OS_ALL,
+  OS_MAC,
+} from 'core/constants';
+import {
+  USER_AGENT_OS_MAC,
+  USER_AGENT_OS_WINDOWS,
+} from 'core/reducers/api';
+import {
+  fakeI18n,
+} from 'tests/unit/helpers';
+import {
+  createFakeAddon,
+  fakePlatformFile,
+} from 'tests/unit/amo/helpers';
+
+describe(__filename, () => {
+  let permissionUtils;
+
+  beforeEach(() => {
+    permissionUtils = new PermissionUtils(fakeI18n());
+  });
+
+  describe('getCurrentPermissions', () => {
+    const createAddon = (platforms = [OS_MAC, OS_ALL]) => {
+      const files = [];
+      for (const platform of platforms) {
+        files.push(
+          {
+            ...fakePlatformFile,
+            platform,
+            permissions: [platform],
```

Comment:
I did it to make the code more concise, but I see what you mean. The function being tested doesn't really care that the array of strings contains permissions - as far as it's concerned it's just an array of strings. So the fact that we're using an invalid permission string doesn't really matter to the test, but I agree it is a bit confusing. To make the test work I need each platform to return something different in the array of strings so that I can verify I'm getting the correct array back. The simplest thing to do, without adding extra code, was to just use the platform name as the permission string. That way I can create it in the loop, and I know what to expect in the test. 

I have added some extra code so it can use valid permission strings, and I can see how it makes the test less confusing. :)